### PR TITLE
Relnotes v2.1

### PR DESCRIPTION
**Pull request by erwango**
V2.1 Release notes update for soc/arm/stm32 and shields sections

### CODE REVIEW

**Review by ioannisg — COMMENTED** [review]
can you see if this overlaps partially with #20901 ??

**Comment by erwango** [comment]
> can you see if this overlaps partially with #20901 ??

Checked. It doesn't.

**Review by ioannisg — APPROVED** [review]

File: doc/releases/release-notes-2.1.rst [hunk 1]
```diff
@@ -28,6 +28,18 @@ Boards & SoC Support
...
+
+  .. rst-class:: rst-columns
+
+     * ssd1306_128x64
+     * ssd1306_128x32
+     * link_board_eth
+     * adafruit_2_8_tft_touch_v2
+     * waveshare_epaper
+     * dfrobot_can_bus_v2_0
+     * x_nucleo_idb05a1
```
Comment (MaureenHelm) [inline doc/releases/release-notes-2.1.rst]: please sort them in alpha order

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.1.rst (hunk 1 above)
Comment (erwango) [reply doc/releases/release-notes-2.1.rst]: Done, txs @MaureenHelm

**Review by MaureenHelm — APPROVED** [review]

**Comment by ioannisg** [comment]
Can you rebase this @erwango ?

**Pull request merged**